engine/release-notes/index.md

```diff
@@ -13,896 +13,164 @@ redirect_from:
+- Added new options-setters to the Go API client: `WithTLSClientConfigFromEnv()`, `WithHostFromEnv()`, and `WithVersionFromEnv()`. [moby/moby#42224](https://github.com/moby/moby/pull/42224)
+- Added generation of shell command completion through `docker completion` subcommand. [docker/cli#3429](https://github.com/docker/cli/pull/3429)
+- API: Added a `Swarm` header to `GET /_ping` and `HEAD /_ping`, allowing single-request detection of Swarm support. [moby/moby#42064](https://github.com/moby/moby/pull/42064)
+- API: Added a `signal` parameter to `POST /containers/{id}/stop` and `POST /containers/{id}/restart` to set the signal used. [moby/moby#43206](https://github.com/moby/moby/pull/43206)
+- API: Added a `CreateMountPoint` parameter to `POST /containers/create`. [moby/moby#43484](https://github.com/moby/moby/pull/43484)
+- API: Added a `shared-size` parameter to `GET /images/json` to enable shared-size computation of images. [moby/moby#42531](https://github.com/moby/moby/pull/42531)
+- API: Added a `type` parameter to `GET /system/df`, to control what object types to are considered when computing disk usage. [moby/moby#42559](https://github.com/moby/moby/pull/42559)
+- systemd: Start `docker.service` after `time-set.target`. [moby/moby#43107](https://github.com/moby/moby/pull/43107)
+
+### Removed
+
+- Removed support for reading configuration from `~/.dockercfg`. [docker/cli#2504](https://github.com/docker/cli/pull/2504)
+
+  This location has been deprecated since 1.7.0.
+
+- Removed the `-g` and `--graph` daemon options in favor of `--data-root`. [docker/cli#3739](https://github.com/docker/cli/pull/3739)
+
+  These options have been hidden and deprecated since 17.05.
+
+- Require explicit configuration to use deprecated storage drivers, and don't automatically select them when upgrading. [moby/moby#43378](https://github.com/moby/moby/pull/43378)
+- Removed deprecated support for `overlay` and `overlay2` storage drivers on backing filesystems without `d_type` support. [moby/moby#43472](https://github.com/moby/moby/pull/43472)
+- Removed the deprecated `overrideKernelCheck` option from the `overlay2` storage driver. [moby/moby#44279](https://github.com/moby/moby/pull/44279)
+- Removed support for the deprecated `io.containerd.runtime.v1.linux` OCI runtime. [moby/moby#43695](https://github.com/moby/moby/pull/43695)
+- Removed LCOW (Linux Containers on Windows). [moby/moby#42451](https://github.com/moby/moby/pull/42451), [moby/moby#42499](https://github.com/moby/moby/pull/42499), [moby/moby#42506](https://github.com/moby/moby/pull/42506), [moby/moby#42511](https://github.com/moby/moby/pull/42511), [moby/moby#42520](https://github.com/moby/moby/pull/42520), [moby/moby#42683](https://github.com/moby/moby/pull/42683), [moby/moby#42684](https://github.com/moby/moby/pull/42684), [moby/moby#42685](https://github.com/moby/moby/pull/42685), [moby/moby#43187](https://github.com/moby/moby/pull/43187)
```

Comment:
If we link to the deprecated page, we could also consider linking to the relevant section on that page (instead of all the pull requests); in this case, for example, it would be https://docs.docker.com/engine/deprecated/#linux-containers-on-windows-lcow-experimental

Reply:
And for a full list of pull requests / changes included in the release, we can (somewhere) provide links to either;

The milestones;

- https://github.com/docker/cli/milestone/51?closed=1
- https://github.com/moby/moby/milestone/91?closed=1

Or overview of merged PRs filtered by milestone;

- https://github.com/docker/cli/pulls?q=is%3Apr+is%3Amerged+milestone%3A23.0.0
- https://github.com/moby/moby/pulls?q=is%3Apr+is%3Amerged+milestone%3A23.0.0

Reply:
I started adding links for each item to the relevant section in the deprecated.md, but it doesn't contain all of the entries, so in that case only some of the items in this list will have a link to a deprecated section.